src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3578,6 +3578,7 @@ void ride_construction_toolupdate_construct(ScreenCoordsXY screenCoords)
                 break;
 
             _currentTrackBegin.z -= 8;
+            // FIXME: Unexpected & LOCATION_NULL. This will be hit on any negative value
```

Comment:
Does the check (for negative values) make sense? I can imagine that you'd want to stop a loop once you go under zero.